File: ta-java/ShoutyReportJob.java
import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.List;
import org.w3c.dom.Document;
import org.w3c.dom.Element;

class ShoutyReportJob {
    public static void main(String[] args) throws IOException {
        String path = args[0];
        List<MileageClaim> mileageClaims = readMileageClaims(path);

        ShoutyReportProcessor job = new ShoutyReportProcessor(mileageClaims);
        writeEcoStatReport(job.process());
    }

/*
    private static RevenueProvider createRevenueProvider() {
        if (System.getenv().get("FAKE_INITIALISATION_DATA") != null) {
            return new FakeRevenueProvider(
                System.getenv().get("FAKE_INITIALISATION_DATA"));
        }

        return new ProductionRevenueProvider();
    }
*/

    private static List<MileageClaim> readMileageClaims(String claimsPath) throws IOException {
        List<MileageClaim> mileageClaims = new ArrayList<>();
        BufferedReader reader = new BufferedReader(new FileReader(claimsPath));
        String line;
        while ((line = reader.readLine()) != null) {
            String[] values = line.split(",");
            mileageClaims.add(new MileageClaim(values[0], Integer.parseInt(values[1]), Integer.parseInt(values[2])));
        }
        return mileageClaims;
    }


    private static void writeEcoStatReport(List<EcoStat> results) {
        Document doc = XmlHelper.newDocument();
        Element ecoReport = doc.createElement("ecoReport");
        doc.appendChild(ecoReport);

        for (EcoStat stat : results) {
            Element node = doc.createElement("ecoStat");
            node.setAttribute("SalespersonName", stat.salesPersonName);
            node.setAttribute("RevenuePerMile", String.valueOf(stat.revenuePerMile));
            ecoReport.appendChild(node);
        }

        XmlHelper.write(doc, "report.xml");
    }
}

File: ta-csharp/ShoutyReportJob.cs
using System.IO;
using System.Collections.Generic;
using System.Xml;

class ShoutyReportJob
{
    public static void Main(string[] args)
    {
        var path = args[0];
        var mileageClaims = ReadMileageClaims(path);

        var job = new ShoutyReportProcessor(mileageClaims);
        WriteEcoStatReport(job.Process());
    }

        /*
        private static IRevenueProvider CreateStatsService()
        {
            if (System.Environment.GetEnvironmentVariable("FAKE_INITIALISATION_DATA") != null)
            {
                return new FakeStatsService(
                    System.Environment.GetEnvironmentVariable("FAKE_INITIALISATION_DATA"));
            }

            return new ProductionRevenueProvider(new ShoutyStatsService());
        }
        */

    private static IList<MileageClaim> ReadMileageClaims(string claimsPath)
    {
        var mileageClaims = new List<MileageClaim>();
        using (var reader = new StreamReader(claimsPath))
        {
            while (!reader.EndOfStream)
            {
                var values = reader.ReadLine().Split(',');
                mileageClaims.Add(new MileageClaim(values[0], int.Parse(values[1]), int.Parse(values[2])));
            }
        }
        return mileageClaims;
    }

    private static void WriteEcoStatReport(List<EcoStat> results)
    {

        XmlDocument reportXml = new XmlDocument();
        reportXml.LoadXml("<?xml version=\"1.0\" encoding=\"UTF-8\"?><ecoReport/>");
        foreach (var ecoStat in results)
        {
            XmlElement node = reportXml.CreateElement(string.Empty, "ecoStat", string.Empty);
            node.SetAttribute("SalespersonName", ecoStat.SalesPersonName);
            node.SetAttribute("RevenuePerMile", System.Convert.ToString(ecoStat.RevenuePerMile));
            reportXml.DocumentElement.AppendChild(node);
        }

        reportXml.Save("report.xml");
    }
}
